common/src/api/external/mod.rs

```diff
@@ -935,6 +935,21 @@ impl DiskState {
     }
 }
 
+/*
+ * SNAPSHOTS
+ */
+
+/// Client view of a Snapshot
+#[derive(ObjectIdentity, Clone, Debug, Deserialize, Serialize, JsonSchema)]
+pub struct Snapshot {
+    #[serde(flatten)]
+    pub identity: IdentityMetadata,
+
+    pub project_id: Uuid,
+    pub disk_id: Uuid,
+    pub size: ByteCount,
+}
+
```

Comment:
would be cool to put this in `views.rs` if possible! we never finished moving things because the networking things were hard to tease apart. might be worth asking someone to poke at that as an onboarding project, it would be a good intro to the omicron architecture

Reply:
Done - gonna do this for the [images PR](https://github.com/oxidecomputer/omicron/pull/749) too in a sec

Reply:
BTW, would you be willing to file a bug with an explanation of that move, if one doesn't exist already?

Reply:
boom https://github.com/oxidecomputer/omicron/issues/388

Reply:
omg back in the days of milestone 3 - a different era